internal/controller/nonadminbackup_controller.go

```diff
@@ -97,11 +144,279 @@ func (r *NonAdminBackupReconciler) Reconcile(ctx context.Context, req ctrl.Reque
+	if err != nil {
+		// Log error if multiple DeleteBackupRequest objects are found
+		logger.Error(err, findSingleVDBRError, uuidString, veleroBackupNameUUID)
+		return false, err
+	}
+
+	if deleteBackupRequest == nil {
+		// Build the delete request for VeleroBackup created by NAC
+		deleteBackupRequest = builder.ForDeleteBackupRequest(r.OADPNamespace, "").
+			BackupName(veleroBackup.Name).
+			ObjectMeta(
+				builder.WithLabels(
+					velerov1.BackupNameLabel, label.GetValidName(veleroBackup.Name),
+					velerov1.BackupUIDLabel, string(veleroBackup.UID),
+					constant.NabOriginNameUUIDLabel, veleroBackupNameUUID,
+				),
+				builder.WithLabelsMap(function.GetNonAdminLabels()),
+				builder.WithAnnotationsMap(function.GetNonAdminBackupAnnotations(nab.ObjectMeta)),
+				builder.WithGenerateName(veleroBackup.Name+"-"),
+			).Result()
+
+		// Use CreateRetryGenerateName for retry logic in creating the delete request
+		if err := function.CreateRetryGenerateName(r.Client, ctx, deleteBackupRequest); err != nil {
+			logger.Error(err, "Failed to create delete request for VeleroBackup", "BackupName", veleroBackup.Name)
+			return false, err
+		}
+		logger.V(1).Info("Request to delete backup submitted successfully", "BackupName", veleroBackup.Name)
+		nab.Status.VeleroDeleteBackupRequest = &nacv1alpha1.VeleroDeleteBackupRequest{
+			NameUUID:  veleroBackupNameUUID,
+			Namespace: r.OADPNamespace,
+			Name:      deleteBackupRequest.Name,
+		}
+	}
+	// Ensure that the NonAdminBackup's NonAdminBackupStatus is in sync
+	// with the DeleteBackupRequest. Any required updates to the NonAdminBackup
+	// Status will be applied based on the current state of the DeleteBackupRequest.
+	updated := updateNonAdminBackupDeleteBackupRequestStatus(&nab.Status, deleteBackupRequest)
+	if updated {
+		if err := r.Status().Update(ctx, nab); err != nil {
+			logger.Error(err, "Failed to update NonAdminBackup Status after DeleteBackupRequest reconciliation")
+			return false, err
+		}
+		logger.V(1).Info("NonAdminBackup DeleteBackupRequest Status updated successfully")
+	}
+	return false, nil // Continue so initNabDeletion can initialize deletion of a NonAdminBackup object
+}
+
+// deleteVeleroBackupAndDeleteBackupRequestObjects deletes both the VeleroBackup and any associated
+// DeleteBackupRequest objects for a given NonAdminBackup when force deletion is requested.
+//
+// Parameters:
+//   - ctx: Context for managing request lifetime
+//   - logger: Logger instance
+//   - nab: NonAdminBackup object
+//
+// Returns:
+//   - bool: whether to requeue (always false)
+//   - error: any error encountered during deletion
+func (r *NonAdminBackupReconciler) deleteVeleroBackupAndDeleteBackupRequestObjects(ctx context.Context, logger logr.Logger, nab *nacv1alpha1.NonAdminBackup) (bool, error) {
+	if nab.Status.VeleroBackup == nil || nab.Status.VeleroBackup.NameUUID == constant.EmptyString {
```

Comment:
would also remove this `if`

Reply:
If we do and for some reason those are nil, we have a controller panic. I can easily replicate this behavior by not enabling velero controller and creating delete request on NAB. This way nab.StatusVeleroBackup is never set and we got panic here.